Bug fix in Ethos runtime backend

limintang opened the pull request:
Summary: Cortex-M is 32-bit, so pointer type in the runtime backend is also 32-bit. If the address >= 0x80000000, direct cast to uint64_t treates it as signed integer and sign extends it to 0xFFFFFFFFXXXXXXXX, which causes address validity check in Ethos driver to fail. First cast it to unsigned type uintptr_t then cast to uint64_t avoids this issue.

Differential Revision: D71658384

limintang commented:
@pytorchbot label "topic: not user facing"

digantdesai reviewed (approved):
Thank you!

zingo reviewed (changes requested):
Thanks for the fix, good find. The fix is really good but there is just a small lintrunner error that needs to be fixed to keep HEAD clean if its OK🙏
If not I can also create a new PR tomorrow with fix for both this and the lint problem, no problem. Thanks for the help.

limintang committed:
Bug fix in Ethos runtime backend (#9517)

Summary:

Cortex-M is 32-bit, so pointer type in the runtime backend is also 32-bit. If the address >= 0x80000000, direct cast to uint64_t treates it as signed integer and sign extends it to 0xFFFFFFFFXXXXXXXX, which causes address validity check in Ethos driver to fail. First cast it to unsigned type uintptr_t then cast to uint64_t avoids this issue.

Reviewed By: digantdesai

Differential Revision: D71658384
